### PR TITLE
Refactor of scenery elements

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Similar to the banner and entrance PR's

WIP

### CODE REVIEW

File: src/openrct2/ride/track_design.c [hunk 1]
```diff
@@ -648,8 +648,8 @@ static sint32 track_design_place_scenery(rct_td6_scenery_element *scenery_start,
 
 					rct_scenery_entry* small_scenery = get_small_scenery_entry(entry_index);
 					if (!(!(small_scenery->small_scenery.flags & SMALL_SCENERY_FLAG_FULL_TILE) &&
-						(small_scenery->small_scenery.flags & SMALL_SCENERY_FLAG9)) &&
-						(small_scenery->small_scenery.flags & (SMALL_SCENERY_FLAG9 | SMALL_SCENERY_FLAG24 | SMALL_SCENERY_FLAG25)))
+						(small_scenery->small_scenery.flags & SMALL_SCENERY_FLAG_DIAGONAL)) &&
+						(small_scenery->small_scenery.flags & (SMALL_SCENERY_FLAG_DIAGONAL | SMALL_SCENERY_FLAG_HALF_SPACE | SMALL_SCENERY_FLAG_THREE_QUARTERS)))
```
Comment (Broxzier) [inline src/openrct2/ride/track_design.c]: `flags` is being checked for `SMALL_SCENERY_FLAG_DIAGONAL` one line up as well.

**Comment by IntelOrca** [comment]
I am not sure where @duncanspumpkin got to with this, but we could merge at least the first 3 commits from this PR as they are mostly just naming unnamed flags.

**Comment by duncanspumpkin** [comment]
I can finish this off on the weekend

File: src/openrct2/windows/top_toolbar.c [hunk 2]
```diff
@@ -1253,14 +1253,14 @@ static void sub_6E1F34(sint16 x, sint16 y, uint16 selected_scenery, sint16* grid
...
-            if (!(scenery->small_scenery.flags & SMALL_SCENERY_FLAG4)){
+            if (!(scenery->small_scenery.flags & SMALL_SCENERY_FLAG_ROTATABLE)){
                 rotation = util_rand() & 0xFF;
             }
 
             rotation -= get_current_rotation();
             rotation &= 0x3;
 
-            // Also places it in lower but think thats for clobbering
+            // Also places it in lower but think thats for clobering
```
Comment (duncanspumpkin) [inline src/openrct2/windows/top_toolbar.c]: undo

File: src/openrct2/windows/tile_inspector.c [hunk 3]
```diff
@@ -1721,7 +1721,7 @@ static void window_tile_inspector_paint(rct_window *w, rct_drawpixelinfo *dpi)
             // Ride
             sint16 rideId = mapElement->properties.track.ride_index;
             rct_ride *ride = get_ride(rideId);
-            rct_string_id rideType = RideNaming[ride->type].name;
+            rct_string_id rideType = STR_RIDE_NAME_SPIRAL_ROLLER_COASTER + ride->type;
```
Comment (duncanspumpkin) [inline src/openrct2/windows/tile_inspector.c]: undo

File: src/openrct2/windows/scenery.c [hunk 4]
```diff
@@ -905,39 +905,34 @@ void window_scenery_tooltip(rct_window* w, rct_widgetindex widgetIndex, rct_stri
...
     if (tabSelectedSceneryId != -1) {
         if (tabSelectedSceneryId < 0x100) {
-            if (!(gWindowSceneryPaintEnabled & 1)) {
+            if (!(gWindowSceneryPaintEnabled & 1))
                 window_scenery_widgets[WIDX_SCENERY_BUILD_CLUSTER_BUTTON].type = WWT_FLATBTN;
-            }
 
             rct_scenery_entry* sceneryEntry = get_small_scenery_entry(tabSelectedSceneryId);
-            if (sceneryEntry->small_scenery.flags & SMALL_SCENERY_FLAG4) {
+            if (sceneryEntry->small_scenery.flags & SMALL_SCENERY_FLAG_ROTATABLE) {
```
Comment (duncanspumpkin) [inline src/openrct2/windows/scenery.c]: undo everything apart from this

File: src/openrct2/windows/top_toolbar.c [hunk 5]
```diff
@@ -1523,7 +1523,7 @@ static void sub_6E1F34(sint16 x, sint16 y, uint16 selected_scenery, sint16* grid
 
         z /= 2;
 
-        // Also places it in lower but think thats for clobbering
+        // Also places it in lower but think thats for clobering
```
Comment (duncanspumpkin) [inline src/openrct2/windows/top_toolbar.c]: undo

File: src/openrct2/windows/top_toolbar.c [hunk 6]
```diff
@@ -1551,9 +1551,6 @@ static void window_top_toolbar_scenery_tool_down(sint16 x, sint16 y, rct_window
     if (gWindowSceneryPaintEnabled & 1) {
         repaint_scenery_tool_down(x, y, widgetIndex);
         return;
-    } else if (gWindowSceneryEyedropperEnabled) {
```
Comment (duncanspumpkin) [inline src/openrct2/windows/top_toolbar.c]: undo

File: src/openrct2/windows/top_toolbar.c [hunk 7]
```diff
@@ -1748,7 +1745,7 @@ static void window_top_toolbar_scenery_tool_down(sint16 x, sint16 y, rct_window
 
         gGameCommandErrorTitle = STR_CANT_POSITION_THIS_HERE;
         game_command_callback = game_command_callback_place_banner;
-        game_do_command(gridX, flags, gridY, parameter_2, GAME_COMMAND_PLACE_BANNER, parameter_3, gWindowSceneryPrimaryColour);
+        game_do_command(gridX, flags, gridY, parameter_2, GAME_COMMAND_PLACE_BANNER, parameter_3, 0);
```
Comment (duncanspumpkin) [inline src/openrct2/windows/top_toolbar.c]: undo

File: src/openrct2/world/map_animation.c [hunk 8]
```diff
@@ -213,7 +213,7 @@ static bool map_animation_invalidate_small_scenery(sint32 x, sint32 y, sint32 ba
                     peep->action = PEEP_ACTION_CHECK_TIME;
                     peep->action_frame = 0;
                     peep->action_sprite_image_offset = 0;
-                    peep_update_current_action_sprite_type(peep);
+                    sub_693B58(peep);
```
Comment (duncanspumpkin) [inline src/openrct2/world/map_animation.c]: undo

File: src/openrct2/paint/map_element/scenery.c [hunk 9]
```diff
@@ -214,27 +214,24 @@ void scenery_paint(uint8 direction, sint32 height, rct_map_element* mapElement)
...
                 }
                 sub_98199C(image_id, x_offset, y_offset, boxlength.x, boxlength.y, boxlength.z - 1, height, boxoffset.x, boxoffset.y, boxoffset.z, rotation);
-            } else {
-                if (entry->small_scenery.flags & SMALL_SCENERY_FLAG16) {
-                    // nothing
-                }
-                sint32 esi = gCurrentTicks;
-                if (!(entry->small_scenery.flags & SMALL_SCENERY_FLAG22)) {
+            } else if (entry->small_scenery.flags & SMALL_SCENERY_FLAG_HAS_FRAME_OFFSETS)
+         {
```
Comment (Gymnasiast) [inline src/openrct2/paint/map_element/scenery.c]: Alignment

File: src/openrct2/world/small_scenery.cpp [hunk 10]
```diff
@@ -0,0 +1,489 @@
...
+        return MONEY32_UNDEFINED;
+    }
+    cost = entry->small_scenery.removal_price * 10;
+
+    gCommandExpenditureType = RCT_EXPENDITURE_TYPE_LANDSCAPING;
+    gCommandPosition.x = x + 16;
+    gCommandPosition.y = y + 16;
+    gCommandPosition.z = baseHeight * 8;
+
+    if (!(flags & GAME_COMMAND_FLAG_GHOST) && game_is_paused() && !gCheatsBuildInPauseMode) {
```
Comment (Gymnasiast) [inline src/openrct2/world/small_scenery.cpp]: All braces should go on their own lines as per the coding style.
There are more occurrences in this file, but I won't leave a line note at every single instance.

File: src/openrct2/paint/map_element/scenery.c [hunk 11]
```diff
@@ -214,27 +214,25 @@ void scenery_paint(uint8 direction, sint32 height, rct_map_element* mapElement)
...
                 }
                 sub_98199C(image_id, x_offset, y_offset, boxlength.x, boxlength.y, boxlength.z - 1, height, boxoffset.x, boxoffset.y, boxoffset.z, rotation);
-            } else {
-                if (entry->small_scenery.flags & SMALL_SCENERY_FLAG16) {
-                    // nothing
-                }
-                sint32 esi = gCurrentTicks;
-                if (!(entry->small_scenery.flags & SMALL_SCENERY_FLAG22)) {
+            }
+            else if (entry->small_scenery.flags & SMALL_SCENERY_FLAG_HAS_FRAME_OFFSETS)
```
Comment (Gymnasiast) [inline src/openrct2/paint/map_element/scenery.c]: You changed the logic here. Previously, the if was inside the else block and did nothing, now you made the whole block dependent on it.

Comment (duncanspumpkin) [inline src/openrct2/paint/map_element/scenery.c]: Yes that was on purpose. The previous logic didn't make sense. In theory the old logic could access an invalid pointer. It doesn't actually make a difference as only a corrupted . dat would hit this code wrong.

**Pull request merged**